include/sessions.hpp

```diff
@@ -362,6 +362,10 @@ struct UserSession
             {
                 userSession->userRole = *thisValue;
             }
+            else if (element.key() == "is_configure_self_only")
+            {
+                userSession->isConfigureSelfOnly = (*thisValue == "true");
```

Comment:
what is the default value for userSession->isConfigureSelfOnly? I have not seen the code for initializing the default value.

Reply:
The route /xyz/openbmc_project/user/root/action/SetPassword is handled by the /xyz/<path> handler.  Here is how I verified that: I added debug statements to both handlers, and found the /xyz/<path> was the only handler being invoked.  I tried changing the order these routes were declared in, that is, moving /xyz/openbmc_project/user/root/action/SetPassword ahead of /xyz/<path> in openbmc_dbus_rest.hpp; that had no effect.  Then finally I deleted the /xyz/openbmc_project/user/root/action/SetPassword handler and tested again.  The SetPassword API works correctly.

Here is the control flow when a client invokes POST /xyz/openbmc_project/user/root/action/SetPassword:
- Handler for BMCWEB_ROUTE(app, "/xyz/<path>").methods(..., "POST"_method, ...) gets control
- This handler performs a custom privilege check which was added by this PR.
- calls handleDBusUrl()
- calls handleAction()
- if (methodName == "SetPassword") calls handleActionSetPassword()
which was previously added to OP940.00 (the base release): https://github.com/ibm-openbmc/bmcweb/blame/OP940/include/openbmc_dbus_rest.hpp#L1493

Reply:
Field UserSession.isConfigureSelfOnly gets its value when the UserSession is constructed.  there are two ways a UserSession is constructed.

1. UserSesion objects are created and added to the SessionStore by SessionStore.generateUserSession() which requires the caller to pass in the configureSelfOnly argument.  There are 3 such call sites:
- include/token_authorization_middleware.hpp - performBasicAuth()
- include/token_authorization_middleware.hpp - handler for POST /login
- redfish-core/lib/redfish_sessions.hpp - SessionCollection.doPost (the handler for /redfish/v1/SessionService/Sessions/)
In each case, the caller has gotten the correct information from pamAuthenticateUser to know if the password is expired.  That value is passed in as the configureSelfOnly argument which then becomes the value of the isConfigureSelfOnly field.

2. UserSession objects are created and added to the SessionStore by UserSession::fromJson().  The value is set if the JSON is_configure_self_only field is present.
The UserSession serializer is in include/sessions.hpp - adl_serializer::to_json.  BMCWeb beginning with OP940.01 will serialize the isConfigureSelfOnly field as the is_configure_self_only member.  Previous versions will not have this field.
If the is_configure_self_only field is not present, UserSession.isConfigureSelfOnly will be created with an indeterminate value.  Although it is likely nobody would be using stored OP940.00 session after an upgrade to OP940.01, I'll change the code to default the value to false. <-- bug fix.

Reply:
>  I'll change the code to default the value to false. <-- bug fix.
Thanks for the detailed explanation, so you will fix the userSession->isConfigureSelfOnly to false.